bot.py

```diff
@@ -712,6 +719,65 @@ async def clear_queue(self, message, _command_content):
         self._stop()
         await message.add_reaction(MusicBot.REACTION_EMOJI)
 
+    async def remove_song_from_deque(self, message, command_content):
+        """
+        Removes a specific song from the queue, based on its position therein
+        """
+        if len(self.media_deque) == 0:
+            await message.channel.send(":unamused: The queue is empty...")
+            return
+
+        arg = command_content.split(maxsplit=1)[0]
+
+        range_regex = re.compile(r"(\d+)-(\d+)")
+        range_regex_match = range_regex.match(arg)
+        if range_regex_match:
+            await self.remove_range_from_deque(message, range_regex_match)
+            return
+
+        if arg.lower() == "first":
+            song_index = 0
+        elif arg.lower() == "last":
+            song_index = -1
+        elif arg.isnumeric():
+            song_index = int(arg) - 1
```

Comment:
Variable initialization isn't necessary here, as all cases where `song_index` isn't assigned are cought by the `else` block and returned. Maybe this makes you uneasy because you're used to statically typed languages? 😂

Reply:
Yes, you are correct. Python can sometimes be :ghost: